Lib/glyphsLib/builder.py

```diff
@@ -371,6 +374,49 @@ def set_custom_params(ufo, parsed=None, data=None, misc_keys=(), non_info=()):
             ufo.lib[GLYPHS_PREFIX + name] = value
 
 
+def parse_glyphs_filter(filter_str):
+    """Parses glyphs custom filter string into a dict object that
+       ufo2ft can consume.
+
+        Reference:
+            ufo2ft: https://github.com/googlei18n/ufo2ft
+            Glyphs 2.3 Handbook July 2016, p184
+
+        Args:
+            filter_str - a string of glyphs app filter
+
+        Return:
+            A dictionary contains the structured filter.
+            Return None if parse failed.
+    """
+    elements = filter_str.split(';')
+
+    if len(elements) == 0 or elements[0] == '':
+        logger.warn('Failed to parse glyphs filter, expecting a filter name: \
```

Comment:
The only possibility of elements[0] being an empty string is that the value of the custom parameter is empty or it starts with a semicolon; in both cases it's invalid and it should not pass unnoticed. The user should fix it in the source file. But I don't mind if you decide to keep it as warning.